.github/workflows/shared-build-and-test.yaml

```diff
@@ -28,6 +28,13 @@ jobs:
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
+      
+      - name: Checkout uid2-shared-actions repo
+        uses: actions/checkout@v4
+        with:
+          ref: kcc-UID2-3858-verify-no-of-tests-run
```

Comment:
Change to v3 before merging

Reply:
Is the checkout of the repo necessary? Is it possible to use a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/avoiding-duplication) instead?

Reply:
It was required for testing. Keep it here as Abu is doing some refactoring laster on so that we don't need to specify the branch name every where when we are testing in the future.